settings/src/components/numeric-control.js

```diff
@@ -13,27 +13,47 @@ import { useRef, useCallback } from '@wordpress/element';
  * using the underlying `input[type="number"]`, which has some accessibility issues.
  *
  * @param props
+ * @param props.autoComplete
+ * @param props.pattern
+ * @param props.title
+ * @param props.onChange
+ * @param props.onFocus
+ * @param props.onKeyDown
+ * @param props.onKeyUp
+ * @param props.index
+ * @param props.value
+ * @param props.maxLength
+ * @param props.required
  * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#accessibility
  * @see https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/
  * @see https://stackoverflow.com/a/66759105/450127
  */
-export default function NumericControl( props ) {
-	const { autoComplete, pattern, title, onChange, onKeyDown, index, value, maxLength, required } =
-		props;
-
+export default function NumericControl( {
+	autoComplete,
+	pattern,
+	title,
+	onChange,
+	onFocus,
+	onKeyDown,
+	onKeyUp,
+	index,
+	value,
+	maxLength,
+	required,
+} ) {
 	const inputRef = useRef( null );
 
-	const handleChange = useCallback(
+	const createHandler = ( handler ) => ( event ) =>
 		// Most callers will only need the value, so make it convenient for them.
-		( event ) => onChange && onChange( event.target.value, event, index, inputRef ),
-		[]
-	);
+		handler && handler( event.target.value, event, index, inputRef.current );
```

Comment:
I don't think the consumer needs to know about the ref, we can just pass the `current` value back